### ISSUE TITLE
different encoding used in oc10 and ocis 

### ISSUE THREAD

**Issue by SwikritiT** [owncloud/ocis#4676]
## Steps to reproduce

1 .As einstein create a folder foo/ and share it to user moss
2. As user moss accept the share
3. As user marie create a folder foo and share it to user moss
4. As moss accept the share
5. As user moss `PROPFIND` the shares folder
```console
curl -u moss:vista -k -X PROPFIND 'https://host.docker.internal:9200/remote.php/webdav/Shares' -H"DEPTH:1" | xmllint --format - 
```

## Expected behavior

In case of `oc10` for second share with same name `/core/remote.php/webdav/Shares/foo%20(2)` such encoding is followed only the space is encoded not the brackets 

```console 
curl -u moss:vista -k -X PROPFIND 'http://localhost/core/remote.php/webdav/Shares' -H"DEPTH:1" | xmllint --format - 
```

```console
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/core/remote.php/webdav/Shares/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 27 Sep 2022 06:05:21 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>0</d:quota-used-bytes>
        <d:quota-available-bytes>-3</d:quota-available-bytes>
        <d:getetag>"8557c636e2d80cf39ac362dee1651cc0"</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/core/remote.php/webdav/Shares/foo/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 27 Sep 2022 06:04:18 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>0</d:quota-used-bytes>
        <d:quota-available-bytes>-3</d:quota-available-bytes>
        <d:getetag>"63329262418f8"</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/core/remote.php/webdav/Shares/foo%20(2)/</d:href>
    <d:propstat>
      <d:prop>
        <d:getlastmodified>Tue, 27 Sep 2022 06:05:21 GMT</d:getlastmodified>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <d:quota-used-bytes>0</d:quota-used-bytes>
        <d:quota-available-bytes>-3</d:quota-available-bytes>
        <d:getetag>"633292a11c85f"</d:getetag>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>
```

## Actual behavior
In case of OCIS the brackets are encoded too `/remote.php/webdav/Shares/foo%20%281%29/`

```
curl -u moss:vista -k -X PROPFIND 'https://host.docker.internal:9200/remote.php/webdav/Shares' -H"DEPTH:1" | xmllint --format - 
``` 
```
<?xml version="1.0"?>
<d:multistatus xmlns:s="http://sabredav.org/ns" xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/webdav/Shares/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668!a0ca6a90-a365-4782-871e-d44447bbc668</oc:id>
        <oc:fileid>a0ca6a90-a365-4782-871e-d44447bbc668$a0ca6a90-a365-4782-871e-d44447bbc668!a0ca6a90-a365-4782-871e-d44447bbc668</oc:fileid>
        <oc:spaceid>a0ca6a90-a365-4782-871e-d44447bbc668</oc:spaceid>
        <d:getetag>"118c53dd590287af6261e9a09bbad1f5"</d:getetag>
        <oc:permissions/>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <oc:size>0</oc:size>
        <d:getlastmodified>Tue, 27 Sep 2022 05:45:07 GMT</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/webdav/Shares/foo%20%281%29/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>1284d238-aa92-42ce-bdc4-0b0000009157$f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c!b790bac1-9a0f-41c3-8368-a3f14057105b</oc:id>
        <oc:fileid>1284d238-aa92-42ce-bdc4-0b0000009157$f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c!b790bac1-9a0f-41c3-8368-a3f14057105b</oc:fileid>
        <oc:spaceid>f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c</oc:spaceid>
        <oc:name>foo</oc:name>
        <d:getetag>"118c53dd590287af6261e9a09bbad1f5"</d:getetag>
        <oc:permissions>SR</oc:permissions>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <oc:size>0</oc:size>
        <d:getlastmodified>Tue, 27 Sep 2022 05:45:07 GMT</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
  <d:response>
    <d:href>/remote.php/webdav/Shares/foo/</d:href>
    <d:propstat>
      <d:prop>
        <oc:id>1284d238-aa92-42ce-bdc4-0b0000009157$4c510ada-c86b-4815-8820-42cdf82c3d51!6e111560-976f-4a5b-b1ab-433d84945f9e</oc:id>
        <oc:fileid>1284d238-aa92-42ce-bdc4-0b0000009157$4c510ada-c86b-4815-8820-42cdf82c3d51!6e111560-976f-4a5b-b1ab-433d84945f9e</oc:fileid>
        <oc:spaceid>4c510ada-c86b-4815-8820-42cdf82c3d51</oc:spaceid>
        <oc:name>foo</oc:name>
        <d:getetag>"f034c275526cddd5a80704ca6b568ca8"</d:getetag>
        <oc:permissions>SR</oc:permissions>
        <d:resourcetype>
          <d:collection/>
        </d:resourcetype>
        <oc:size>0</oc:size>
        <d:getlastmodified>Tue, 27 Sep 2022 05:44:26 GMT</d:getlastmodified>
        <oc:favorite>0</oc:favorite>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>

```

## Setup

Please describe how you started the server and provide a list of relevant environment variables or configuration files.

<details>
<p>

```console
OCIS_XXX=somevalue
OCIS_YYY=somevalue
PROXY_XXX=somevalue
```

</p>
</details>

**Comment by butonic** [comment]
@C0rby wanted to use the default path encoding for href in https://github.com/cs3org/reva/blob/7e300c41608401c96515cd317a371941c688ff58/internal/http/services/owncloud/ocdav/net/net.go#L75-L77

now, href is defined as [`<!ELEMENT href (#PCDATA)>`](http://www.webdav.org/specs/rfc4918.html#ELEMENT_href) and PCDATA aka as parsed chartacter data in the context of an XML document means we only have to encode `<` and `&` as we did for preperties with https://github.com/cs3org/reva/pull/3240

but that will not even encode ` ` as `%20`, which we should do. That is why [sabredav has its own encoding](https://github.com/sabre-io/http/blob/bb27d1a8c92217b34e778ee09dcf79d9a2936e84/lib/functions.php#L369-L379) which we were reimplementing before: https://github.com/cs3org/reva/blob/05267c411caaeed135506d2c75bbcfb3a5f31fb7/internal/http/services/owncloud/ocdav/net/net.go#L90-L103

In summary, percent encoding `(` and `)` is one possible way of encoding urls. IMO the testsuite should urldecode the href before comparing the content.
